node/tests/SharedTests.ts

```diff
@@ -12240,6 +12240,159 @@ export function runBaseTests(config: {
         },
         config.timeout,
     );
+
+    describe.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "update_connection_password_%p",
+        (protocol) => {
+            const NEW_PASSWORD = "new_password";
+            const WRONG_PASSWORD = "wrong_password";
+            /**
+             * Test replacing the connection password without immediate re-authentication.
+             * Verifies that:
+             * 1. The client can update its internal password
+             * 2. The client remains connected with current auth
+             * 3. The client can reconnect using the new password after server password change
+             * Currently, this test is only supported for cluster mode,
+             * since standalone mode dont have retry mechanism.
```

Comment:
vice versa - only in standalone (according to our client configs)
Update in tests for other clients too

Reply:
So the there isn't a lot of use for it in standalone, but this is the first building block of wider implemantetion of error handling of auth. At the next phase the user will be able to set a task calling to auth base on preconfigured time and wider error handling will be added internally in the core using the saved password.

In the meantime, if the user have an IAM auth and he dont want to recreate the client every 12 hours, plus we dont want to create an odd state when we have saved in conf one password while the user use different one, the API ofer a cleaner way to perform re-auth needed while changing the configuration. When better error handling will be added, the user would not need to change his implemantetion, and the enhancing would happen under the hood with no strong noise.

So i see two reasons why to have it added, and no reason to not have it, since it doesnt brake anything and it does exactly what it says it does. so i dont see why not.